src/optimum/nvidia/weights/base.py

```diff
@@ -66,3 +158,25 @@ def allocate_model(conf: ModelConfig, sharding: ShardingConfig, dtype: DataType,
         :return:
         """
         raise NotImplementedError()
+
+    @classmethod
+    def from_safetensors(
+        cls,
+        paths: List[Union[str, "PathLike"]],
+        model: "Module",
+        config: ModelConfig,
+        builder_config: BuilderConfig,
+        qconfig: QuantizationConfig,
+        sharding_config: "ShardingConfig",
+    ) -> "Module":
+        if not isinstance(model, cls.TENSORRT_LLM_MODEL_CLASS):
```

Comment:
`from_safetensors` is abstracted here, hence the need for `TENSORRT_LLM_MODEL_CLASS` defined in `LlamaWeightAdapter`, `WhisperDecoderWeightAdapter`, `WhisperEncoderWeightAdapter`.